torch/nn/functional.py

```diff
@@ -1997,6 +1997,10 @@ def hardsigmoid(input: Tensor, inplace: bool = False) -> Tensor:
 
 Applies a linear transformation to the incoming data: :math:`y = xA^T + b`.
 
+This opperation supports 2-D :attr:`weight` with :ref:`sparse layout<sparse-docs>`
```

Comment:
Technically, some cases with sparse `input` would also be supported. However this was the operative case we were working toward so I outlined it explicitly. 

cc @cpuhrsch